[docs][cloud-provider-yandex] Add docs about Deckhouse Kubernetes Platform integration with Yandex Cloud Managed Services (#7209)
deckhouse/deckhouse: modules/030-cloud-provider-yandex/docs/README.md

- Provisions disks in Yandex Cloud using the `CSI storage` component.
- Registers with the [node-manager](../../modules/040-node-manager/) module so that [YandexInstanceClasses](cr.html#yandexinstanceclass) can be used when creating the [NodeGroup](../../modules/040-node-manager/cr.html#nodegroup).
- Enables the necessary CNI plugin (using the [simple bridge](../../modules/035-cni-simple-bridge/)).

## Интеграция с Yandex Cloud

### Настройка групп безопасности

> При создании [облачной сети](https://cloud.yandex.ru/ru/docs/vpc/concepts/network#network), Yandex Cloud создаёт [группу безопасности](https://cloud.yandex.ru/ru/docs/vpc/concepts/security-groups) по умолчанию для всех сетей, в том числе и для сети кластера Deckhouse Kubernetes Platform. Группа безопасности по умолчанию содержит правила разрешающие любой трафик в любом направлении (входящий и исходящий) и применяется для всех подсетей в рамках облачной сети, если на объект (интерфейс ВМ) явно не назначена другая группа безопасности. Вы можете изменить правила группы безопасности по умолчанию, если вам необходимо контролировать трафик в вашем кластере.
> 

Reviewer: ```suggestion
При создании [облачной сети](https://cloud.yandex.ru/ru/docs/vpc/concepts/network#network), Yandex Cloud создаёт [группу безопасности](https://cloud.yandex.ru/ru/docs/vpc/concepts/security-groups) по умолчанию для всех сетей, в том числе и для сети кластера Deckhouse Kubernetes Platform. Группа безопасности по умолчанию содержит правила разрешающие любой трафик (входящий и исходящий) и применяется для всех подсетей в облачной сети, если на объект (интерфейс ВМ) не назначена другая группа безопасности. Существует возможность изменения правил группы безопасности по умолчанию, если необходимо контролировать трафик в пользовательском кластере.
```

> При создании [облачной сети](https://cloud.yandex.ru/ru/docs/vpc/concepts/network#network), Yandex Cloud создаёт [группу безопасности](https://cloud.yandex.ru/ru/docs/vpc/concepts/security-groups) по умолчанию для всех сетей, в том числе и для сети кластера Deckhouse Kubernetes Platform. Группа безопасности по умолчанию содержит правила разрешающие любой трафик в любом направлении (входящий и исходящий) и применяется для всех подсетей в рамках облачной сети, если на объект (интерфейс ВМ) явно не назначена другая группа безопасности. Вы можете изменить правила группы безопасности по умолчанию, если вам необходимо контролировать трафик в вашем кластере.
> 

Reviewer: ```suggestion
При создании [облачной сети](https://cloud.yandex.ru/ru/docs/vpc/concepts/network#network), Yandex Cloud создаёт [группу безопасности](https://cloud.yandex.ru/ru/docs/vpc/concepts/security-groups) по умолчанию для всех сетей, в том числе и для сети кластера Deckhouse Kubernetes Platform. Группа безопасности по умолчанию содержит правила разрешающие любой трафик в любом направлении (входящий и исходящий) и применяется для всех подсетей в рамках облачной сети, если на объект (интерфейс ВМ) явно не назначена другая группа безопаsssssssssssssсности. Вы можете изменить правила группы безопасности по умолчанию, если вам необходимо контролировать трафик в вашем кластере.

```

{% alert level="danger" %}

> Do not delete the rule that allows for traffic to pass in any direction before finishing configuring all the other rules for the security group. Doing so may disrupt the performance of the cluster.

Reviewer: ```suggestion
Do not delete the default rules that allow for traffic to pass in any direction before finishing configuring all the other rules for the security group. Doing so may disrupt the performance of the cluster.
```

Reviewer 2: Look good to me

> This section provides general guidelines for setting up a security group. Incorrect configuration of security groups may affect the performance of the cluster. Please consult [security group usage details](https://cloud.yandex.com/en/docs/vpc/concepts/security-groups#security-groups-notes) in Yandex Cloud before using it in production environments.
> 
> {% alert level="danger" %}
> Do not delete the rule that allows for traffic to pass in any direction before finishing configuring all the other rules for the security group. Doing so may disrupt the performance of the cluster.
> {% endalert %}

Reviewer: ```suggestion
{% alert level="danger" %}
Do not delete the rule that allows for traffic to pass in any direction before finishing configuring all the other rules for the security group. Doing so may disrupt the performance of the cluster.
{% endalert %}

This section provides general guidelines for setting up a security group. Incorrect configuration of security groups may affect the performance of the cluster. Please consult [security group usage details](https://cloud.yandex.com/en/docs/vpc/concepts/security-groups#security-groups-notes) in Yandex Cloud before using it in production environments.
```

1. Find out in which cloud network the Deckhouse Kubernetes Platform cluster is running.

>    The network name matches the `prefix` field of the [ClusterConfiguration] resource(../../installing/configuration.html#clusterconfiguration). It can be retrieved using the following command:
> 

Reviewer: ```suggestion
   The network name matches the `prefix` field of the [ClusterConfiguration](../../installing/configuration.html#clusterconfiguration) resource. It can be retrieved using the following command:

```

   ```bash
   kubectl get secrets -n kube-system d8-cluster-configuration -ojson | \
     jq -r '.data."cluster-configuration.yaml"' | base64 -d | grep prefix | cut -d: -f2
   ```

> 1. В консоли Yandex Cloud выберите сервис Virtual Private Cloud и перейдите в раздел *Группы безопасности*. У вас должна отображаться одна группа безопасности с пометкой `Default`.
> 
>     ![Группа безопасности по умолчанию](../../images/030-cloud-provider-yandex/sg-ru-default.png)
> 
> 1. Создайте правила согласно [инструкции Yandex Cloud](https://cloud.yandex.ru/ru/docs/managed-kubernetes/operations/connect/security-groups#rules-internal).
> 
>     ![Правила для группы безопасности](../../images/030-cloud-provider-yandex/sg-ru-rules.png)
> 
> 1. Удалите правило, разрешающее любой **входящий** трафик (на скриншоте выше оно уже удалено), и сохраните изменения.

Reviewer 3: ```suggestion
1. В консоли Yandex Cloud выберите сервис Virtual Private Cloud и перейдите в раздел *Группы безопасности*.  В результате должна отображаться одна группа безопасности с пометкой `Default`.

    ![Группа безопасности по умолчанию](../../images/030-cloud-provider-yandex/sg-ru-default.png)

1. Создайте правила по [инструкции Yandex Cloud](https://cloud.yandex.ru/ru/docs/managed-kubernetes/operations/connect/security-groups#rules-internal).

    ![Правила для группы безопасности](../../images/030-cloud-provider-yandex/sg-ru-rules.png)

1. Удалите правило, разрешающее любой **входящий** трафик (на вышеприведенном скриншоте правило удалено), и сохраните изменения.
```

Приведенную инструкцию следует рассматривать как *Быстрый старт*. Для использования интеграции в продуктивных средах ознакомьтесь со следующими ресурсами:

- [Yandex Lockbox](https://cloud.yandex.ru/ru/docs/lockbox/)
- [Синхронизация с секретами Yandex Lockbox](https://cloud.yandex.ru/ru/docs/managed-kubernetes/tutorials/kubernetes-lockbox-secrets)
- [External Secret Operator](https://external-secrets.io/latest/)

#### Инструкция по развертыванию

1. [Создайте сервисный аккаунт](https://cloud.yandex.com/ru/docs/iam/operations/sa/create), необходимый для работы External Secrets Operator:

   ```shell
   yc iam service-account create --name eso-service-account
   ```

1. [Создайте авторизованный ключ](https://cloud.yandex.ru/ru/docs/iam/operations/authorized-key/create) для сервисного аккаунта и сохраните его в файл:

   ```shell
   yc iam key create --service-account-name eso-service-account --output authorized-key.json
   ```

> 1. [Назначьте](https://cloud.yandex.ru/ru/docs/iam/operations/sa/assign-role-for-sa) сервисному аккаунту [роли](https://cloud.yandex.com/ru/docs/lockbox/security/#service-roles) `lockbox.editor`, `lockbox.payloadViewer` и `kms.keys.encrypterDecrypter` для доступа ко всем секретам каталога:
> 
>    ```shell
>    folder_id=<идентификатор каталога>
>    yc resource-manager folder add-access-binding --id=${folder_id} --service-account-name eso-service-account --role lockbox.editor
>    yc resource-manager folder add-access-binding --id=${folder_id} --service-account-name eso-service-account --role lockbox.payloadViewer
>    yc resource-manager folder add-access-binding --id=${folder_id} --service-account-name eso-service-account --role kms.keys.encrypterDecrypter
>    ```
> 
>    Для более тонкой настройки ознакомьтесь с [управлением доступом в Yandex Lockbox](https://cloud.yandex.com/ru/docs/lockbox/security).
> 

Reviewer: ```suggestion
1. [Назначьте](https://cloud.yandex.ru/ru/docs/iam/operations/sa/assign-role-for-sa) сервисному аккаунту [роли](https://cloud.yandex.com/ru/docs/lockbox/security/#service-roles) `lockbox.editor`, `lockbox.payloadViewer` и `kms.keys.encrypterDecrypter` для доступа ко всем секретам каталога:

   ```shell
   folder_=<идентификатор каталога>
   yc resource-manager folder add-access-binding --id=${folder_id} --service-account-name eso-service-account --role lockbox.editor
   yc resource-manager folder add-access-binding --id=${folder_id} --service-account-name eso-service-account --role lockbox.payloadViewer
   yc resource-manager folder add-access-binding --id=${folder_id} --service-account-name eso-service-account --role kms.keys.encrypterDecrypter
   ```

   Для более тонкой настройки ознакомьтесь с [управлением доступом в Yandex Lockbox](https://cloud.yandex.com/ru/docs/lockbox/security).

```

         auth:
           authorizedKeySecretRef:
             name: sa-creds
             key: key'
   ```

   Where:
   - `sa-creds` — the name of the `Secret` that contains the authorized key. This secret should show up after the Helm Chart has been installed.
   - `key` — the name of the key in the `.data` field of the secret above.

#### Checking that everything works as expected

1. Check the status of the External Secrets Operator and the secrets store you created:

   ```shell
   $ kubectl -n external-secrets get po
   NAME                                                READY   STATUS    RESTARTS   AGE
   external-secrets-55f78c44cf-dbf6q                   1/1     Running   0          77m
   external-secrets-cert-controller-78cbc7d9c8-rszhx   1/1     Running   0          77m
   external-secrets-webhook-6d7b66758-s7v9c            1/1     Running   0          77m

   $ kubectl -n external-secrets get secretstores.external-secrets.io 
   NAME           AGE   STATUS
   secret-store   69m   Valid
   ```

1. [Create](https://cloud.yandex.com/en/docs/lockbox/operations/secret-create) a Yandex Lockbox secret with the following parameters:

    - **Name** — `lockbox-secret`.

>     - **Kay** — enter the non-confidential identifier `password`.

Reviewer: ```suggestion
    - **Key** — enter the non-confidential identifier `password`.
```
